plugins/inputs/sqlserver/sqlserverqueries.go

```diff
@@ -0,0 +1,1133 @@
+		N'DBMIRROR_DBM_EVENT', N'DBMIRROR_EVENTS_QUEUE', N'DBMIRROR_WORKER_QUEUE',
+		N'DBMIRRORING_CMD', N'DIRTY_PAGE_POLL', N'DISPATCHER_QUEUE_SEMAPHORE',
+		N'EXECSYNC', N'FSAGENT', N'FT_IFTS_SCHEDULER_IDLE_WAIT', N'FT_IFTSHC_MUTEX',
+		N'HADR_CLUSAPI_CALL', N'HADR_FILESTREAM_IOMGR_IOCOMPLETION', N'HADR_LOGCAPTURE_WAIT',
+		N'HADR_NOTIFICATION_DEQUEUE', N'HADR_TIMER_TASK', N'HADR_WORK_QUEUE',
+		N'KSOURCE_WAKEUP', N'LAZYWRITER_SLEEP', N'LOGMGR_QUEUE',
+		N'MEMORY_ALLOCATION_EXT', N'ONDEMAND_TASK_QUEUE',
+		N'PARALLEL_REDO_WORKER_WAIT_WORK',
+		N'PREEMPTIVE_HADR_LEASE_MECHANISM', N'PREEMPTIVE_SP_SERVER_DIAGNOSTICS',
+		N'PREEMPTIVE_OS_LIBRARYOPS', N'PREEMPTIVE_OS_COMOPS', N'PREEMPTIVE_OS_CRYPTOPS',
+		N'PREEMPTIVE_OS_PIPEOPS','PREEMPTIVE_OS_GENERICOPS', N'PREEMPTIVE_OS_VERIFYTRUST',
+		N'PREEMPTIVE_OS_DEVICEOPS',
+		N'PREEMPTIVE_XE_CALLBACKEXECUTE', N'PREEMPTIVE_XE_DISPATCHER',
+		N'PREEMPTIVE_XE_GETTARGETSTATE', N'PREEMPTIVE_XE_SESSIONCOMMIT',
+		N'PREEMPTIVE_XE_TARGETINIT', N'PREEMPTIVE_XE_TARGETFINALIZE',
+		N'PWAIT_ALL_COMPONENTS_INITIALIZED', N'PWAIT_DIRECTLOGCONSUMER_GETNEXT',
+		N'QDS_PERSIST_TASK_MAIN_LOOP_SLEEP',
+		N'QDS_ASYNC_QUEUE',
+		N'QDS_CLEANUP_STALE_QUERIES_TASK_MAIN_LOOP_SLEEP', N'REQUEST_FOR_DEADLOCK_SEARCH',
+		N'RESOURCE_QUEUE', N'SERVER_IDLE_CHECK', N'SLEEP_BPOOL_FLUSH', N'SLEEP_DBSTARTUP',
+		N'SLEEP_DCOMSTARTUP', N'SLEEP_MASTERDBREADY', N'SLEEP_MASTERMDREADY',
+		N'SLEEP_MASTERUPGRADED', N'SLEEP_MSDBSTARTUP', N'SLEEP_SYSTEMTASK', N'SLEEP_TASK',
+		N'SLEEP_TEMPDBSTARTUP', N'SNI_HTTP_ACCEPT', N'SP_SERVER_DIAGNOSTICS_SLEEP',
+		N'SQLTRACE_BUFFER_FLUSH', N'SQLTRACE_INCREMENTAL_FLUSH_SLEEP',
+		N'SQLTRACE_WAIT_ENTRIES',
+		N'WAIT_FOR_RESULTS', N'WAITFOR', N'WAITFOR_TASKSHUTDOWN', N'WAIT_XTP_HOST_WAIT',
+		N'WAIT_XTP_OFFLINE_CKPT_NEW_LOG', N'WAIT_XTP_CKPT_CLOSE',
+		N'XE_BUFFERMGR_ALLPROCESSED_EVENT', N'XE_DISPATCHER_JOIN',
+		N'XE_DISPATCHER_WAIT', N'XE_LIVE_TARGET_TVF', N'XE_TIMER_EVENT',
+		N'SOS_WORK_DISPATCHER','RESERVED_MEMORY_ALLOCATION_EXT')
+	AND waiting_tasks_count > 0
+	AND wait_time_ms > 100;
+`
+
+const sqlServerRequests string = `
+SET NOCOUNT ON; 
+SELECT  blocking_session_id into #blockingSessions FROM sys.dm_exec_requests WHERE blocking_session_id != 0
+create index ix_blockingSessions_1 on #blockingSessions (blocking_session_id)
+SELECT	
+'sqlserver_requests' AS [measurement]
+, REPLACE(@@SERVERNAME,'\',':') AS [sql_instance]
+, s.session_id
+, ISNULL(r.request_id,0) as request_id
+, DB_NAME(s.database_id) as session_db_name
+, COALESCE(r.status,s.status) AS status
+, COALESCE(r.cpu_time,s.cpu_time) AS cpu_time_ms
+, COALESCE(r.total_elapsed_time,s.total_elapsed_time) AS total_elapsed_time_ms
+, COALESCE(r.logical_reads,s.logical_reads) AS logical_reads
+, COALESCE(r.writes,s.writes) AS writes
+, r.command
+, r.wait_time as wait_time_ms
+, r.wait_type
+, r.wait_resource
+, r.blocking_session_id
+, s.program_name
+, s.host_name
+, s.nt_user_name
+, s.open_transaction_count  AS open_transaction
```

Comment:
For 2008 and 2008R2 compatibility the same can be obtained with
```SQL
,open_transaction =
        CASE
            WHEN r.session_id IS NOT NULL THEN r.open_transaction_count
            ELSE (SELECT COUNT(*) FROM sys.dm_tran_session_transactions AS t WHERE t.session_id = s.session_id)
        END
```
Thanks @spaghettidba

Reply:
This wasn't something I changed, haven't committed so far, can discuss under different PR. Having a count(*) from open transactions  for every request is a cheap operation ( think of 5000 active requests). This results in a nested loop join executed for every transaction. SQL 2008/R2 is past extended support from a SQL Server perspective, at some point trying to get parity for everything new isn't going to be possible and will complicate existing queries.

Reply:
This has been changed in the 45a83b9, the priority has been given to the sessions table (s) before thew column was picked from the request table (r). (which to me still makes sense tbh, since the table is detailed "by request" and not by session. 
Probably the best solution is to use ```COALESCE(r.open_transaction_count, s.open_transaction_count) AS [open_transaction_count]```, by doing so we get the data per-request, but if there is no request then it makes sense to fetch it from the session table.

We can revert it back to ```r.___``` and get the data from the requests, or I can split the query in 2 versions using dynamic SQL in order to manage this better.

I'm aware of the fact that SQL 2008/R2 is past extended support, and I hope to don't see it around anymore.... but for now it's still around...
So I'd like to keep the full compatibility at least for the V2 of the queries, but I agree on the fact that a certain point the support for it should be stopped

Reply:
These are totally 2 different discussions
a. I get the Requests v/s Sessions argument for open_transactions, that is a trivial change and shouldn't be different or if ti is the session count should override anyways as tran is still open
b. what I was commenting on is the fact that open_transactions isn't there before 2012 and hence the select to get the count from dm_tran_session_transactions is more expensive.. Either way this wasn't a change

I am not sure where you are seeing it changed in https://github.com/influxdata/telegraf/commit/45a83b9ab487b3ac5bfafc24ce30257bce91fa75 I just took a look a the line says   
, s.open_transaction_count  AS open_transaction

Reply:
In this case I would point to solution "a", so just get the column form requests instead of sessions (for compatibility reasons, otherwise the query explodes on SQL 2008). 

Just abort the idea of the subquery on "dm_tran_session_transactions" (it was not a great idea), as you said it's heavier and it's not worth running it just to keep compatibility with SQL 2008 and R2

Reply:
I ended up changing to dynamic SQL  to get appropriate column. Changing to r.open_transactions means that even for versions after 2008R2 you will not get the open tran count for an Idle session ( think of a session that has an open transaction and no request).